lib/class-sp-admin.php

```diff
@@ -308,23 +323,29 @@ public function settings_page() {
 							<strong><?php esc_html_e( 'Because this site has a large number of posts, this may take a long time to index.', 'searchpress' ); ?></strong>
 						<?php endif ?>
 						<?php esc_html_e( "Exactly how long indexing will take will vary on a number of factors, like the server's CPU and memory, connection speed, current traffic, average post size, and associated terms and post meta.", 'searchpress' ); ?>
-						<?php esc_html_e( 'SearchPress will be inactive during indexing if you choose to "Flush the data and update the mapping".', 'searchpress' ); ?>
+						<?php
+						if ( $this->allow_flushing ) : ?>
+							<?php esc_html_e( 'SearchPress will be inactive during indexing if you choose to "Flush the data and update the mapping".', 'searchpress' ); ?>
+						<?php endif; ?>
 					</p>
 
 					<form method="post" action="<?php echo esc_url( admin_url( 'admin-post.php' ) ); ?>">
 						<input type="hidden" name="action" value="sp_full_sync" />
-						<p>
-							<label for="sp_flush_data">
-								<input type="checkbox"
-									name="sp_flush_data"
-									id="sp_flush_data"
-									value="1"
-									<?php checked( 0 === SP_Sync_Manager()->count_posts_indexed() ); ?>
-								/>
-								<?php esc_html_e( 'Flush the data and update the mapping', 'searchpress' ); ?>
-							</label>
-							<span class="explanation"><?php esc_html_e( 'This will wipe the data currently in the Elasticsearch index and rebuild it from scratch. This is necessary in order to udpate the mapping.', 'searchpress' ); ?></span>
-						</p>
+						<?php
+						if ( $this->allow_flushing ) : ?>
+							<p>
+								<label for="sp_flush_data">
+									<input type="checkbox"
+										name="sp_flush_data"
+										id="sp_flush_data"
+										value="1"
+										<?php checked( 0 === SP_Sync_Manager()->count_posts_indexed() ); ?>
+									/>
+									<?php esc_html_e( 'Flush the data and update the mapping', 'searchpress' ); ?>
+								</label>
+								<span class="explanation"><?php esc_html_e( 'This will wipe the data currently in the Elasticsearch index and rebuild it from scratch. This is necessary in order to udpate the mapping.', 'searchpress' ); ?></span>
```

Comment:
heh, just noticed I had a typo in this message. when you merge this to a release branch, can you fix `udpate`?